packages/common/src/services/audius-backend/solana.ts

```diff
@@ -757,3 +713,210 @@ export const createVersionedTransaction = async (
+  analyticsFields,
+  signer
+}: {
+  sdk: AudiusSdk
+  mint: PublicKey
+  connection: Connection
+  amount: number
+  ethWallet: string
+  destinationWallet: PublicKey
+  track: CommonStoreContext['analytics']['track']
+  make: CommonStoreContext['analytics']['make']
+  analyticsFields: any
+  signer?: Keypair
+}) => {
+  let isCreatingTokenAccount = false
+  try {
+    const instructions: TransactionInstruction[] = []
+
+    const destination = getAssociatedTokenAddressSync(mint, destinationWallet)
+
+    try {
+      await getAccount(connection, destination)
+    } catch (e) {
+      // Throws if token account doesn't exist or account isn't a token account
+      isCreatingTokenAccount = true
+      console.debug(e)
+      console.debug(
+        'Associated token account',
+        destination.toBase58(),
+        'does not exist. Creating w/ transfer...'
+      )
+
+      // Historically, the token account was created in a separate transaction
+      // after swapping USDC to SOL via Jupiter and funded via the root wallet.
+      // This is no longer the case. Reusing the same Amplitude events anyway.
+      await track(
+        make({
+          eventName: Name.WITHDRAW_USDC_CREATE_DEST_TOKEN_ACCOUNT_START,
+          ...analyticsFields
+        })
+      )
+      const payerKey = await sdk.services.solanaRelay.getFeePayer()
+      const createAtaInstruction =
+        createAssociatedTokenAccountIdempotentInstruction(
+          payerKey,
+          destination,
+          destinationWallet,
+          mint
+        )
+      instructions.push(createAtaInstruction)
+    }
+
+    const secpTransferInstruction =
+      await sdk.services.claimableTokensClient.createTransferSecpInstruction({
+        auth: sdk.services.auth,
+        amount,
+        ethWallet,
+        mint,
+        destination,
+        instructionIndex: instructions.length
+      })
+    instructions.push(secpTransferInstruction)
+
+    const transferInstruction =
+      await sdk.services.claimableTokensClient.createTransferInstruction({
+        ethWallet,
+        mint,
+        destination
+      })
+    instructions.push(transferInstruction)
+
+    if (signer) {
+      const memoInstruction = new TransactionInstruction({
+        keys: [
+          {
+            pubkey: signer.publicKey,
+            isSigner: true,
+            isWritable: true
+          }
+        ],
+        programId: MEMO_PROGRAM_ID,
+        data: Buffer.from(PREPARE_WITHDRAWAL_MEMO_STRING)
+      })
+      instructions.push(memoInstruction)
+    }
+
+    const transaction =
+      await sdk.services.claimableTokensClient.buildTransaction({
+        instructions
+      })
+
+    if (signer) {
+      transaction.sign([signer])
+    }
+
+    const signature = await sdk.services.claimableTokensClient.sendTransaction(
+      transaction
+    )
+
+    if (isCreatingTokenAccount) {
+      await track(
+        make({
+          eventName: Name.WITHDRAW_USDC_CREATE_DEST_TOKEN_ACCOUNT_SUCCESS,
```

Comment:
Are there any other events we should be sending for this process? Or do we not track the transfer itself as an analytics event?

Reply:
I *think* the analytics events are different depending on the caller, and we record at the callsites.

Reply:
https://github.com/AudiusProject/audius-protocol/blob/c1822b1eb856cf60d412d3c7c1e1c52d1b605ae8/packages/web/src/store/application/ui/withdraw-usdc/sagas.ts#L251-L255 yep